library/std/src/io/error.rs

```diff
@@ -852,15 +852,15 @@ impl Error {
         }
     }
 
-    /// Attempt to downcast the inner error to `E` if any.
+    /// Attempt to downcast the custom boxed error to `E`.
     ///
-    /// If this [`Error`] was constructed via [`new`] then this function will
-    /// attempt to perform downgrade on it, otherwise it will return [`Err`].
+    /// If this [`Error`] when this contains a custom boxed error,
+    /// then it would attempt downcasting on the boxed error,
+    /// otherwise it will return [`Err`].
     ///
-    /// If the downcast succeeds, it will return [`Ok`], otherwise it will also
-    /// return [`Err`].
+    /// If the custom boxed error has the same type as `E`, it will return [`Ok`],
+    /// otherwise it will also return [`Err`].
```

Comment:
Can you also mention here that this is meant to be a convenience routine for calling [`Box<dyn Error + Sync + Send>::downcast`](https://doc.rust-lang.org/std/error/trait.Error.html#method.downcast-2) on the custom boxed error?

Reply:
Yes, I've added them to the doc